limiter.go

```diff
@@ -72,12 +72,35 @@ func (l *RateLimiter) OnLimit(w http.ResponseWriter, r *http.Request, key string
 	currentWindow := time.Now().UTC().Truncate(l.windowLength)
 	ctx := r.Context()
 
-	limit := l.requestLimit
-	if val := getRequestLimit(ctx); val > 0 {
-		limit = val
+	setHeader(w, l.headers.Reset, fmt.Sprintf("%d", currentWindow.Add(l.windowLength).Unix()))
+
+	limit, ok := getRequestLimit(ctx)
+	if !ok {
+		limit = l.requestLimit
+	}
+	// If the limit is set to 0, we are always over limit
+	if limit == 0 {
+		return true
```

Comment:
we should be sending some headers back

X-RateLimit-Limit: 0
X-RateLimit-Remaining: 0